File: search-insert-position/search_insert_position.go
func searchInsert(nums []int, target int) int {
    for k, v := range nums {
        if v > target {
            return 0
        } else if v == target {
            return k
        } else if (k == len(nums) - 1) || (v < target && nums[k+1] > target) {
            return k + 1
        }
    }
    return 0
}

File: search-insert-position/search_insert_position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        for i in range(len(nums)):
            if nums[i] > target:
                return 0
            elif nums[i] == target:
                return i
            elif i == len(nums)-1 or nums[i] < target and nums[i+1] > target:
                return i+1